frontend/packages/kubevirt-plugin/integration-tests/tests/mocks/vmBuilderPresets.ts

```diff
@@ -0,0 +1,124 @@
+import { deepFreeze } from '../utils/utils';
+import { VMBuilder } from '../models/vmBuilder';
+import * as Combinatorics from 'js-combinatorics';
+import { Flavor, Workload, OperatingSystem, ProvisionSource } from '../utils/constants/wizard';
+import { VMBuilderDataGenerationConfig, VMBuilderData } from '../types/vm';
+import { testName } from '@console/internal-integration-tests/protractor.conf';
+import {
+  provisionSources,
+  rootDisk,
+  multusNetworkInterface,
+  containerRootDisk,
+  flavorConfigs,
+  getDiskToCloneFrom,
+} from './mocks';
+import { VirtualMachine } from '../models/virtualMachine';
+import { VMTemplateBuilder } from '../models/vmtemplateBuilder';
+import { VirtualMachineTemplate } from '../models/virtualMachineTemplate';
+
+const generateBuilders = (baseBuilder, groupIndex, keys, values) => {
```

Comment:
can I ask for some documentation here? :)

Reply:
Yep, while writing the doc I realized that the way the logic was separated was pretty user-hostile. So changed it a bit and added a doc